### PR TITLE
Faster UI - React Tweaks

### PR DESCRIPTION
**Pull request by ngotchac**
Related to #3240 

Mainly pre-render accounts list so the click on the Accounts tab instantaneously render something instead of waiting for all the accounts to be rendered.

Smarter Account Summary Component and better tabs (used to be rendered around 30 times whenever the tabs switched).

Also added React Perf in dev mode : in JS console:

* `Perf.start()`
* `Perf.stop()`
* `Perf.printWasted()`

This will show in a table all the useless re-renders.

### CODE REVIEW

File: js/src/ui/ParityBackground/parityBackground.js [hunk 1]
```diff
@@ -16,26 +16,17 @@
...
 import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
 
 class ParityBackground extends Component {
-  static contextTypes = {
-    muiTheme: PropTypes.object.isRequired
-  }
-
   static propTypes = {
+    style: PropTypes.object.isRequired,
```
Comment (jacogr) [inline js/src/ui/ParityBackground/parityBackground.js]: Care to explain why we are passing this in while it is available on context?

Comment (ngotchac) [inline js/src/ui/ParityBackground/parityBackground.js]: This is just to not re-compute the background from the given seed for each render (it's computed once, at least while the seed isn't changing, by the Redux `mapStateToProps` method.)

Comment (ngotchac) [inline js/src/ui/ParityBackground/parityBackground.js]: This is just to not re-compute the background from the given seed for each render (it's computed once, at least while the seed isn't changing, by the Redux mapStateToProps method.)

Comment (jacogr) [inline js/src/ui/ParityBackground/parityBackground.js]: Couldn't it just have been done in `componentWillMount`?

I think the single calculation is great, just concerned about "special cases" where we start doing things inconsistently.

Comment (derhuerst) [inline js/src/ui/ParityBackground/parityBackground.js]: > So we'd have to choose whether to do it in the component or have a state-less component with props managed by Redux.

A general note:

Both Redux & Mobx encourage users to split React components into "dumb"/presentational and "smart"/behavioural. We'd have a smart component connecting the store/state to the React components tree and lots of "dumb", pure components beneath.

However, memoizing the background computation and returning a value that can be checked for equality by React should work in terms of performance.

**Comment by jacogr** [comment]
CI tests still seems to have issues, will get back to this once recovered.

**Comment by ngotchac** [comment]
Yep, `no space left on device`...

**Pull request merged**